app/charts/shared/brush/index.tsx

```diff
@@ -106,6 +106,11 @@ export const BrushTime = () => {
         xEnd,
       ]);
 
+      // Need to use current state as the function is not updated during brushing
+      // and the local state accessed here is not up to date. This leads to
+      // making a dispatch on each brush move, which makes the animations laggy
+      // and generally shouldn't happen.
+      const { from, to } = useInteractiveFiltersStore.getState().timeRange;
```

Comment:
Reading the comment on the commit, I am wondering why are we using an old brush component ? Why is there several versions of the brush ?

Here if I am correct, the brush is "disconnected" (in the sense that if something external to the brush changes from/to, the brush will not be re-rendered) from the interactive filters store, because this is the one controlling it ?

Reply:
I wasn't 100% sure about the cause, but the problem was that we accessed the old interactive filters state inside the `brushed` function. I guess this function is "frozen" once user starts interacting with the brush (that's what I meant with the old brush 😅), and when the interactive filters state updates, this is not detected, because we are using the old callback. By changing the logic to the line you commented, we are able to always access up-to-date state.